### ISSUE TITLE
[multihost] allow publishing processed images for each hosts

### ISSUE THREAD

**Issue by razonyang** [gohugoio/hugo#12110]
Image resource's `Permalink` work fine for original image on multihost setup, but produces a wrong link after processing the image, e.g. resize and filters.

### Reproduce Steps

```sh
$ git clone -b processed-image-link git@github.com:razonyang/hugo-testing

$ cd hugo-testing

$ hugo server
```

And then navigated to http://localhost:1313/foo/ and http://localhost:1314/foo/.

The processed images on `1314` page are referenced to `1313`.

### What version of Hugo are you using (`hugo version`)?

<pre>
$ hugo version
hugo v0.123.1-3f8434a62fc91a50f4e5bd43a4435d2d301844ff+extended linux/amd64 BuildDate=2024-02-21T08:17:45Z VendorInfo=snap:0.123.1
</pre>

### Does this issue reproduce with the latest release?

Yes.

**Comment by bep** [comment]
Are you sure this isn't a "shared resource"? There will be cases where you in a multihost setup links from hosta to hostb or vice versa.

**Comment by bep** [comment]
>Would this be considered as a feature request?

Well, I wouldn't call it a bug if it's by design. Could you change the title/label to better reflect what it is.

**Comment by razonyang** [comment]
Thanks for considering this! I changed the title, but have no right to edit the label.

**Comment by razonyang** [comment]
Seems fixed by #12167, related to #12163.
